#!/usr/bin/env python3

PACKAGE = "waypoints_follower_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        Name                Type      Level  Description                      Default   Min   Max
gen.add( "frame_rate" ,      int_t,     1,     "Collecting data rate",            7,     1,  14)
# gen.add( "pub_rate" ,        int_t,     1,     "Publish rate",                    3,     1,  10)
# gen.add( "inference_rate",   int_t,     1,     "Infernce rate",                   3,     1,    10)
gen.add( "wpt_i" ,           int_t,     1,     "Waypoint index",                  5,     0,    16)
gen.add( "smoothen_time" ,   double_t,  1,     "Smoothen time factor",            0.0,   0.0,  1.5)

                 # PkgName  #NodeName         #Prefix for generated .h include file, e.g. ParametersConfig.py
exit(gen.generate(PACKAGE, "pilot_goal_generation_publisher", "Parameters"))